packages/proxy/src/utils.ts

```diff
@@ -26,20 +26,48 @@ async function getFileHash(filePath: string) {
   return createHash("md5").update(fileBuffer).digest("hex");
 }
 
-export async function createFileWatcher(filePath: string, onChange: () => void) {
-  try {
-    const configWatcher = fs.watch(filePath);
-    let previousFileHash = getFileHash(filePath);
-
-    for await (const event of configWatcher) {
-      if (event.eventType !== "change") continue;
-      const newFileHash = getFileHash(filePath);
-      if (previousFileHash !== newFileHash) {
-        onChange();
-        previousFileHash = newFileHash;
+function withResolvers() {
+  let resolve!: (_: unknown) => void;
+  let reject!: (_: unknown) => void;
+  const promise = new Promise((res, rej) => {
+    resolve = res;
+    reject = rej;
+  });
+  return {
+    resolve,
+    reject,
+    promise,
+  };
+}
+
+export function createFileWatcher(filePath: string, onChange: () => void) {
+  const ac = new AbortController();
+  const { promise, resolve } = withResolvers();
+  (async () => {
+    try {
+      const configWatcher = fs.watch(filePath, { signal: ac.signal });
+      let previousFileHash = getFileHash(filePath);
+
+      for await (const event of configWatcher) {
+        if (event.eventType !== "change") continue;
+        const newFileHash = getFileHash(filePath);
+        if (previousFileHash !== newFileHash) {
+          onChange();
+          previousFileHash = newFileHash;
+        }
+      }
+    } catch (err: unknown) {
+      if (err instanceof Error && err.name === "AbortError") {
+        console.log("Stopped file watcher");
+        resolve({});
```

Comment:
why are we resolving with an empty object and not `void`?

Reply:
i brokith the types and bent the knee